### PR TITLE
More tests for sift

### PR DESCRIPTION
**Pull request by Yosshi999**
Issue #1

### CODE REVIEW

**Comment by Yosshi999** [comment]
The test data is: https://github.com/Yosshi999/opencv_extra/commit/6122d8d3375985cfd3685c20f75fbed11e26b587

**Comment by Yosshi999** [comment]
I think this exact test can be failed because the current implementation is not bit-exact.
~I will withdraw the commit `fdb8b3d`.~

If I can't retain the same result after creating bit-exact implementation, I will update the test data in opencv_extra. (that means, withdraw Yosshi999/opencv_extra@6122d8d and re-commit the result)

File: modules/features2d/test/test_sift.cpp [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    CV_SiftTest();
+    ~CV_SiftTest();
+protected:
+    void run(int);
+};
+
+CV_SiftTest::CV_SiftTest() {}
+CV_SiftTest::~CV_SiftTest() {}
+
+void CV_SiftTest::run( int )
```
Comment (terfendail) [inline modules/features2d/test/test_sift.cpp:21]: Sorry for confusing you about regression test for SIFT descriptor. It looks like there is [one](https://github.com/terfendail/opencv/blob/fdb8b3dfa887738f48d742f451b01391066ea2d9/modules/features2d/test/test_descriptors_regression.cpp#L21). It verifies against `opencv_extra/testdata/cv/features2d/descriptor_extractors/descriptor-sift` file.
Could you please check it? If there already is a test against precomputed values additional test becomes redundant.

Comment (Yosshi999) [reply modules/features2d/test/test_sift.cpp:21]: This test case will be passed when 80% of results are in an allowable error.
In contrast commit fdb8b3d is for testing bit-exactness and will be passed only when all results are exactly similar.
So I think we will need this test in implementing bit-exact version but I don't think it is necessary now.

File: modules/features2d/test/test_sift.cpp [hunk 2]
```diff
@@ -0,0 +1,145 @@
+/*M///////////////////////////////////////////////////////////////////////////////////////
+//
+//  IMPORTANT: READ BEFORE DOWNLOADING, COPYING, INSTALLING OR USING.
+//
+//  By downloading, copying, installing or using the software you agree to this license.
+//  If you do not agree to this license, do not download, install,
```
Comment (terfendail) [inline modules/features2d/test/test_sift.cpp]: Could you please use a short licence header
```
// This file is part of OpenCV project.
// It is subject to the license terms in the LICENSE file found in the top-level directory
// of this distribution and at http://opencv.org/license.html.
```

File: modules/features2d/test/test_sift.cpp (hunk 1 above)
Comment (terfendail) [reply modules/features2d/test/test_sift.cpp:21]: I've missed the test threshold. In that case exact test is necessary for future development.

File: modules/features2d/test/test_sift.cpp [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+// warranties of merchantability and fitness for a particular purpose are disclaimed.
+// In no event shall the Intel Corporation or contributors be liable for any direct,
+// indirect, incidental, special, exemplary, or consequential damages
+// (including, but not limited to, procurement of substitute goods or services;
+// loss of use, data, or profits; or business interruption) however caused
+// and on any theory of liability, whether in contract, strict liability,
+// or tort (including negligence or otherwise) arising in any way out of
+// the use of this software, even if advised of the possibility of such damage.
+//
+//M*/
```
Comment (Yosshi999) [inline modules/features2d/test/test_sift.cpp]:
```suggestion
// This file is part of OpenCV project.
// It is subject to the license terms in the LICENSE file found in the top-level directory
// of this distribution and at http://opencv.org/license.html.
```

**Review by terfendail — APPROVED** [review]

**Pull request merged**